### DISCUSSION TITLE
How to apply styling?

### DISCUSSION THREAD

**Discussion by mandolyte** [jgm/djot#47] [Q&A] [answered]
Suppose my document has:
```
Greeting{.heading}
```
then where can I provide, say:
```
heading {
   color: red;
}
```
so that when I create the HTML via pandoc, then this styling information will be included.

I'm sure this is just my limits on how pandoc works, but after trying a couple of things, I thought I'd ask. Thanks for any pointers.

**Comment by jgm** [comment]
With pandoc you can use `--css` to specify a CSS file that can include this.
Or use `--template` with a custom template.
Or use `--include-in-header` with a file containing a `<style>` attribute.

**Reply by mandolyte** [reply]
Then perhaps my command is missing something. Here is my command:
```
pandoc -f ../src/djot-reader.lua \
        --css=styles.css \
        -o jude.html \
        jude.djot
```
Here is styles.css:
```
book {
        color: red;
}

heading {
        color: blue;
}
```
But the html output has no css content, no <syles>. The first two lines are:
```html
<p><span class="book">JUDE</span></p>
<p><span class="heading">Greeting</span></p>
```
So it looks good as far as it goes; but expected a styles tag and maybe a header and body to wrap things.

**Reply by jgm** [reply]
You also need `-s` (`--standalone`) to get a full document with a header.

**Reply by mandolyte** [reply]
The `-s` plus adding leading dots to my css elements did the trick. So for posterity: 
- my script:
```
pandoc -f ../src/djot-reader.lua \
        --css=styles.css \
        --metadata title="Jude" \
        -s \
        -o jude.html \
        jude.djot
```

And `styles.css`:
```
header { color: green; }
.book {
        color: red;
}

.heading {
        color: blue;
}
```

**Comment by mandolyte** [comment, marked as answer]
Confirmed - see above.
